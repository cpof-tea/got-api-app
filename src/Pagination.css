.pagination {
	display: grid;
	grid-auto-flow: column;
	grid-auto-columns: max-content;
	gap: 10px;
}

