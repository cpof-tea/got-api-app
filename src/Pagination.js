import './Pagination.css';

export default function Pagination(props) {
	const { pages, page, pageSize, onChange } = props;

	return pages ? (
		<div className="pagination">
			<span>
				{ page !== 1 ? (
						<button
							type="button"
							onClick={ () => onChange(1, pageSize) }
						>
							First page
						</button>
				) : null }
				{ page !== 1 ? (
						<button
							type="button"
							onClick={ () => onChange(page - 1, pageSize) }
						>
							Previous page
						</button>
				) : null }
				{ page !== pages ? (
						<button
							type="button"
							onClick={ () => onChange(page + 1, pageSize) }
						>
							Next page
						</button>
				) : null }
				{ page !== pages ? (
					<button
						type="button"
						onClick={ () => onChange(pages, pageSize) }
					>
						Last page
					</button>
				) : null }
			</span>
			<span>
				Page { page } of { pages }
			</span>
			<span>
				Items per page&nbsp;
				 <select
					onChange={ (event) => onChange(1, event.currentTarget.value) }
					defaultValue={ pageSize }
				>
					{ [10, 25, 50].map((value) => (
						<option key={ value } value={ value }>{ value }</option>
					)) }
				</select>
			</span>
		</div>
	) : null;
}

