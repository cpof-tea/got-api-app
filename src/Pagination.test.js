import { render, screen } from '@testing-library/react';
import Pagination from './Pagination';

test('pagination render', async () => {
	const { asFragment, rerender } = render(
		<Pagination
			page={ 1 }
			pageSize={ 25 }
			pages={ 3 }
		/>
	);

	expect(asFragment()).toMatchSnapshot();

	rerender(
		<Pagination
			page={ 3 }
			pageSize={ 10 }
			pages={ 3 }
		/>
	);

	expect(asFragment()).toMatchSnapshot();

	rerender(
		<Pagination
			page={ 2 }
			pageSize={ 50 }
			pages={ 3 }
		/>
	);

	expect(asFragment()).toMatchSnapshot();

	rerender(
		<Pagination
			page={ 1 }
			pageSize={ 10 }
			pages={ 1 }
		/>
	);

	expect(asFragment()).toMatchSnapshot();
});

