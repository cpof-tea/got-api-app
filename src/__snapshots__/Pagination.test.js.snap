// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pagination render 1`] = `
<DocumentFragment>
  <div
    class="pagination"
  >
    <span>
      <button
        type="button"
      >
        Next page
      </button>
      <button
        type="button"
      >
        Last page
      </button>
    </span>
    <span>
      Page 1 of 3
    </span>
    <span>
      Items per page 
      <select>
        <option
          value="10"
        >
          10
        </option>
        <option
          selected=""
          value="25"
        >
          25
        </option>
        <option
          value="50"
        >
          50
        </option>
      </select>
    </span>
  </div>
</DocumentFragment>
`;

exports[`pagination render 2`] = `
<DocumentFragment>
  <div
    class="pagination"
  >
    <span>
      <button
        type="button"
      >
        First page
      </button>
      <button
        type="button"
      >
        Previous page
      </button>
    </span>
    <span>
      Page 3 of 3
    </span>
    <span>
      Items per page 
      <select>
        <option
          value="10"
        >
          10
        </option>
        <option
          selected=""
          value="25"
        >
          25
        </option>
        <option
          value="50"
        >
          50
        </option>
      </select>
    </span>
  </div>
</DocumentFragment>
`;

exports[`pagination render 3`] = `
<DocumentFragment>
  <div
    class="pagination"
  >
    <span>
      <button
        type="button"
      >
        First page
      </button>
      <button
        type="button"
      >
        Previous page
      </button>
      <button
        type="button"
      >
        Next page
      </button>
      <button
        type="button"
      >
        Last page
      </button>
    </span>
    <span>
      Page 2 of 3
    </span>
    <span>
      Items per page 
      <select>
        <option
          value="10"
        >
          10
        </option>
        <option
          selected=""
          value="25"
        >
          25
        </option>
        <option
          value="50"
        >
          50
        </option>
      </select>
    </span>
  </div>
</DocumentFragment>
`;

exports[`pagination render 4`] = `
<DocumentFragment>
  <div
    class="pagination"
  >
    <span />
    <span>
      Page 1 of 1
    </span>
    <span>
      Items per page 
      <select>
        <option
          value="10"
        >
          10
        </option>
        <option
          selected=""
          value="25"
        >
          25
        </option>
        <option
          value="50"
        >
          50
        </option>
      </select>
    </span>
  </div>
</DocumentFragment>
`;
